src/Services/Item/Item.BusinessLogic/Services/Implementations/ItemService.cs

```diff
@@ -0,0 +1,237 @@
+    {
+        var item = await _itemRepository.GetByIdAsync(id, token);
+
+        NotFoundException.ThrowIfNull(item);
+
+        var currentUserId = _currentUserService.UserId;
+
+        var currentRole = _currentUserService.Role;
+
+        if (currentUserId == item.UserId ||
+            currentRole == nameof(Role.Administrator) ||
+            currentRole == nameof(Role.Moderator))
+        {
+            await _imageService.DeleteAllAttachedImagesAsync(item.Id, token);
+
+            await _itemRepository.DeleteAsync(item, token);
+
+            return item;
+        }
+        else
+        {
+            throw new ForbiddenException();
+        }
+    }
+
+    public async Task<Item> ChangeStatus(Guid id, UpdateStatusDto updateStatusDto, CancellationToken token = default)
+    {
+        var specification = new ItemWithStatusSpecification(id);
+        var item = await _itemRepository.FirstOrDefaultAsync(specification, token);
+
+        NotFoundException.ThrowIfNull(item);
+
+        var currentStatus = item.Status!;
+
+        var newStatus = await _statusRepository.FirstOrDefaultAsync(x => x.NormalizedName == updateStatusDto.NormalizedName, token);
+
+        NotFoundException.ThrowIfNull(newStatus);
+
+        var currentRole = _currentUserService.Role;
+
+        if (_currentUserService.UserId == item.UserId)
+        {
+            if ((currentStatus.Equals(StatusValues.Active) && newStatus.Equals(StatusValues.Inactive)) ||
+                (currentStatus.Equals(StatusValues.Inactive) && newStatus.Equals(StatusValues.Active)))
+            {
+                item.Status = newStatus;
+            }
+            else
+            {
+                throw new ConflictException(ItemErrorMessages.StatusFailure);
+            }
+        }
+        else if (currentRole == nameof(Role.Administrator) || currentRole == nameof(Role.Moderator))
+        {
+            item.Status = newStatus;
+        }
+        else
+        {
+            throw new ForbiddenException();
+        }
+
+        return await _itemRepository.UpdateAsync(item, token);
+    }
+
+    public async Task<Item> GetByIdAsync(Guid id, CancellationToken token)
+    {
+        var specification = new ItemWithAllSpecification(id);
+        var item = await _itemRepository.FirstOrDefaultAsync(specification, token);
+
+        NotFoundException.ThrowIfNull(item);
+
+        return item;
+    }
+
+    public async Task<PagedList<Item>> GetAsync(ItemFilterQuery filterQuery, CancellationToken token = default)
+    {
+        var specification = new ItemWithAllSpecification();
+        var itemQuery = _itemRepository.GetQueryable(specification);
+
+        if (!string.IsNullOrWhiteSpace(filterQuery.SearchTerm))
+        {
+            itemQuery = itemQuery.Where(x => x.Title.Contains(filterQuery.SearchTerm));
+        }
+
+        if (filterQuery.CityId is not null)
+        {
+            itemQuery = itemQuery.Where(x => x.CityId == filterQuery.CityId);
+        }
```

Comment:
there is a heavy load on the services here, it is better to take it out in the repository. Since returning IQueryable to services is a bad idea